gait3d/gaitanalysis_with_top_cam.py

```diff
@@ -0,0 +1,996 @@
+                    break
+                else:
+                    print(f'Entered paradigm does not match any of the defined paradigms. \nPlease enter one of the following paradigms: {self.valid_paradigms}')
+        else:
+            self.paradigm = paradigm
+        try:
+            int(recording_date)
+            self.recording_date = recording_date
+        except:
+            while True:
+                entered_input = input(f'Recording date for {self.filepath}')
+                try:
+                    int(recording_date)
+                    self.recording_date = recording_date
+                    break
+                except:
+                    print(f'Entered recording date has to be an integer in shape YYMMDD. Example: 220812')       
+
+    
+    def preprocess(self,
+                   marker_ids_to_compute_coverage: List[str]=['TailBase', 'Snout'],
+                   coverage_threshold: float=0.75, 
+                   max_seconds_to_interpolate: float=0.5, 
+                   likelihood_threshold: float=0.5,
+                   marker_ids_to_compute_center_of_gravity: List[str]=['TailBase', 'Snout'],
+                   relative_maze_normalization_error_tolerance: float=0.25
+                   ) -> None:
+        initial_logs_to_add = {'critical_markers_to_compute_coverage': marker_ids_to_compute_coverage,
+                               'coverage_threshold': coverage_threshold,
+                               'max_seconds_to_interpolate': max_seconds_to_interpolate,
+                               'min_likelihood_threshold': likelihood_threshold,
+                               'center_of_gravity_based_on': marker_ids_to_compute_center_of_gravity, 
+                               'relative_error_tolerance_corner_detection': relative_maze_normalization_error_tolerance}
+        window_length = self._get_max_odd_n_frames_for_time_interval(fps = self.fps, time_interval = max_seconds_to_interpolate)
+        marker_ids_to_preprocess = self._get_preprocessing_relevant_marker_ids(df = self.full_df_from_file)
+        smoothed_df = self._smooth_tracked_coords_and_likelihood(marker_ids = marker_ids_to_preprocess, window_length = window_length, polyorder = 3)
+        interpolated_df = self._interpolate_low_likelihood_intervals(df = smoothed_df, marker_ids = marker_ids_to_preprocess, max_interval_length = window_length)
+        interpolated_df_with_cog = self._add_new_marker_derived_existing_markers(df = interpolated_df,
+                                                                                 existing_markers = marker_ids_to_compute_center_of_gravity,
+                                                                                 new_marker_id = 'CenterOfGravity',
+                                                                                 likelihood_threshold = likelihood_threshold)
+        preprocessed_df = self._interpolate_low_likelihood_intervals(df = interpolated_df_with_cog,
```

Comment:
Yes I was aware of this and I also could not really decide on whether I like it or not. I also couldn´t come up with a strong reason of why it would not be valid, so I included it for now to maybe squeeze out some additional percentages for the coverage. But I also opened an [issue](https://github.com/Defense-Circuits-Lab/Gait_Analysis/issues/40) for this, just that we don´t forget about it.